File: apps/frontend/src/api/logs.ts

import api from './client'
import type { DecisionTraceEntry, ReasoningSummary, EvidenceLink, KnowledgeDeltaEntry, LogIndex, LogType, LogCategory } from './types'

export async function getLogIndex(run_id: string): Promise<LogIndex> {
  const r = await api.get(`/logs/${run_id}/index`)
  return r.data
}
export async function getDecisionTrace(run_id: string): Promise<DecisionTraceEntry[]> {
  const r = await api.get(`/logs/${run_id}/decision_trace`)
  return r.data
}
export async function getSummary(run_id: string): Promise<ReasoningSummary> {
  const r = await api.get(`/logs/${run_id}/summary`)
  return r.data
}
export async function getEvidenceLinks(run_id: string): Promise<EvidenceLink[]> {
  const r = await api.get(`/logs/${run_id}/evidence_links`)
  return r.data
}
export async function getKnowledgeDelta(run_id: string): Promise<KnowledgeDeltaEntry[]> {
  const r = await api.get(`/logs/${run_id}/knowledge_delta`)
  return r.data
}
export async function searchLogs(run_id: string, q?: string, category?: LogCategory) : Promise<DecisionTraceEntry[]> {
  const r = await api.get('/logs/search', { params: { run_id, q, category }})
  return r.data
}

File: apps/backend/src/schemas/logs.py
from __future__ import annotations

from typing import List, Literal, Optional
from pydantic import BaseModel, Field
from datetime import datetime

LogCategory = Literal['plan','execution','evidence','governance','audit','metrics','learning']
LogType = Literal['decision_trace','reasoning_summary','evidence_links','knowledge_delta']

class DecisionTraceEntry(BaseModel):
    id: str
    ts: datetime
    gpee_step: Literal['goal','plan','execute','evaluate']
    ooda_step: Literal['observe','orient','decide','act']
    action: str
    score: dict = Field(description='score breakdown, redacted')
    outcome: str = Field(description='short, redacted outcome summary (no raw CoT)')
    category: LogCategory = 'plan'
    tags: List[str] = []

class ReasoningSummary(BaseModel):
    run_id: str
    ts: datetime
    title: str
    bullets: List[str] = Field(description='structured, redacted summary bullets')
    categories: List[LogCategory] = []
    tags: List[str] = []

class EvidenceLink(BaseModel):
    id: str
    ts: datetime
    label: str
    path: str
    artifact_type: str
    category: LogCategory = 'evidence'
    tags: List[str] = []

class KnowledgeDeltaEntry(BaseModel):
    id: str
    ts: datetime
    summary: str
    impact_score: int = Field(ge=0, le=100)
    applies_to: List[str] = Field(description='e.g., persona, policy, graph, dorks')
    category: LogCategory = 'learning'
    tags: List[str] = []

class LogIndex(BaseModel):
    run_id: str
    available: List[LogType]
    counts: dict
